src/extension/inpage-script/index.js

```diff
@@ -5,6 +5,9 @@ import WebLNProvider from "../ln/webln";
 if (document) {
   window.webln = new WebLNProvider();
 
+  const readyEvent = new Event("webln:ready");
+  document.dispatchEvent(readyEvent);
```

Comment:
@reneaaron what do you think about this, is this helpful?

Reply:
Personally I never needed such an event before. 🤔 Do you have any specific use-cases in mind where one would need that event? 

Since app devs are not supposed to trigger any payments without a user interacting (I think it would be strange to trigger `webln.enable()` or any other call on page load, right?) I wonder where one would use this event?

Reply:
I think the main use case would probably be to know if webln is available. A website wants to check if the user supports webln calls or if a fallback must be used or an error must be shown. 

`window.webln` is loaded on `document_end` - when the DOM is loaded. But checking for it on DOMContentLoaded might be still too early.

I just thought to fire such an event does not hurt.

Reply:
Okay, so the app would check for WebLN capabilities right on page load to setup payment components or something. I would have never needed that before but I agree that offering this event doesn't hurt. 👍

Reply:
I also don't care about this one that much.
my use case was more for the `lightning:success` event. that allows to have a normal lightning: protocol link but still get informed if a payment was done (if a user has alby)

Reply:
Yeah, that makes sense. That could potentially allow developers to integrate a very low-impact solution for their existing apps. All they'd have to do would be to add `lightning:` links to their markup without going the full WebLN way of doing things. (and still supporting mobile wallets)

I am not too sure about the naming, though. 
 - `lightning` vs. `webln` vs. `webbtc`: Could this be part of the spec some day? I just don't want to confuse people with different names
 - Maybe it would make sense to be a bit more specific (i.e. `lightning:paymentSuccess`) in case there are other events that might become available in the future?  🤔

Reply:
If you do a call let me know. I think I have quite some things to learn.

Reply:
I just noticed a bug on https://webln.twentyuno.net where the detection of WebLN is run on `load` and users are redirected to an onboarding-page if they don't have WebLN. 

Sometimes it seems the `window.load` event is triggered before the webln provider is injected into the page. I think an event like `webln:ready` could help in such a case 👍  (you'd still need to add a timout for a redirect since non WebLN browsers won't trigger this event). 

I guess there is no way to inject the WebLN provider before the `load` event is fired, right?